chia/rpc/wallet_rpc_api.py

```diff
@@ -3457,161 +3427,116 @@ async def nft_get_info(self, request: dict[str, Any]) -> EndpointResult:
         )
         # This is a bit hacky, it should just come out like this, but this works for this RPC
         nft_info = dataclasses.replace(nft_info, p2_address=p2_puzzle_hash)
-        return {"success": True, "nft_info": nft_info}
+        return NFTGetInfoResponse(nft_info)
 
     @tx_endpoint(push=True)
+    @marshal
     async def nft_add_uri(
         self,
-        request: dict[str, Any],
+        request: NFTAddURI,
         action_scope: WalletActionScope,
         extra_conditions: tuple[Condition, ...] = tuple(),
-    ) -> EndpointResult:
-        wallet_id = uint32(request["wallet_id"])
+    ) -> NFTAddURIResponse:
         # Note metadata updater can only add one uri for one field per spend.
         # If you want to add multiple uris for one field, you need to spend multiple times.
-        nft_wallet = self.service.wallet_state_manager.get_wallet(id=wallet_id, required_type=NFTWallet)
-        uri = request["uri"]
-        key = request["key"]
-        nft_coin_id = request["nft_coin_id"]
-        if nft_coin_id.startswith(AddressType.NFT.hrp(self.service.config)):
-            nft_coin_id = decode_puzzle_hash(nft_coin_id)
+        nft_wallet = self.service.wallet_state_manager.get_wallet(id=request.wallet_id, required_type=NFTWallet)
+        if request.nft_coin_id.startswith(AddressType.NFT.hrp(self.service.config)):
+            nft_coin_id = decode_puzzle_hash(request.nft_coin_id)
         else:
-            nft_coin_id = bytes32.from_hexstr(nft_coin_id)
+            nft_coin_id = bytes32.from_hexstr(request.nft_coin_id)
```

Comment:
someday, a helper for this pattern might be worthwhile.  `:]`  feel free to close this after reading.